### ISSUE TITLE
estimated discount lifetime reward loss function

### ISSUE THREAD

**Issue by sbenthall** [scikit-agent/scikit-agent#22]
Given a block (`DBlock` or `RBlock`), produce a function that represents the estimated discount lifetime reward, such that it can be a loss function for training a neural network.
